docs/migration/draft-v9-migration-guide.md

```diff
@@ -34,6 +34,27 @@
 
 - Deprecated `Request` in favor of `RequestEventData`.
 
+## `@sentry/vue`
+
+- Deprecated `tracingOptions`, `trackComponents`, `timeout`, `hooks` options everywhere other than in the `tracingOptions` option of the `vueIntegration()`.
+  These options should now be set as follows:
+
+  ```ts
+  import * as Sentry from '@sentry/vue';
+
+  Sentry.init({
+    integrations: [
+      Sentry.vueIntegration({
+        tracingOptions: {
+          trackComponents: true,
```

Comment:
the only thing we could think about elevating to root-level config IMHO is `trackComponents`, this seems somewhat important enough to possibly warrant this (and also should not be too confusing). So IMHO we could still allow this specifically to also be passed directly to `init()`, in addition to allowing to pass it to the integration, but no strong feelings.

Reply:
I also admittedly don't have hard feelings towards this. I would err towards not allowing it top-level because it opens the door for things tree-shaking in the future, and less ambiguous docs.

@Lms24 @s1gr1d do you have opinions on whether or not we should still allow `trackComponents` as a top-level `Sentry.init()` option?

Reply:
I'd prefer keeping it integration only because if `Sentry.vueIntegration` is filtered out, then `trackComponents` doesn't do anything.

Reply:
I opened a PR to update the docs. That will maybe highlight a bit better what the config will look like: https://github.com/getsentry/sentry-docs/pull/11917

Reply:
Component tracking options are mostly defined outside the Sentry.init options across our SDKs. I think it's fine to only have them at integration level in Vue. For Nuxt, we probably want to figure out a way of still letting users change things but I'd argue this is a long-term goal. AFAIK, we don't have docs on component tracking in meta framework SDKs at all at the moment.

Reply:
Ah I see how you envisioned it for Nuxt - users add the `vueIntegration` themselves. However, I think that would not work here, as the Nuxt SDK already adds the `vueIntegration` internally to setup the Vue router. This happens here: https://github.com/getsentry/sentry-javascript/blob/472c2288bc184323348665eeef6bd25ba36496af/packages/nuxt/src/runtime/plugins/sentry.client.ts#L56

So we would still have to add something like a `vueOptions` key for the Nuxt SDK `init` as we cannot call the integration twice.

And what about the passed Vue app? I don't see `app: vueApp` in the passed options in the docs code examples 👀

Reply:
> So we would still have to add something like a vueOptions key for the Nuxt SDK init as we cannot call the integration twice.

i think that's fine